float64[3] pos # 3x1 mtx
uint32     id
